### PR TITLE
fix: track bulk add input for Tracked Wizard Modal in localStorage

### LINKED ISSUE #2966: Bug: text field gets cleared out on modal if you click on import and go back
### Describe the bug

Rewatch GIF Recording - 2024-03-18 at 5 00 04 PM

### Steps to reproduce

<stub>

### Browsers

_No response_

### Additional context (Is this in dev or production?)

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by zeucapua**
## Description

Keeps the input for bulk add step for `TrackedRepoWizard` and `TrackedContributorWizard` in local storage so that moving away from the tab or going back and forth in the wizard step.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes #2966 
Relates to #2799 

## Mobile & Desktop Screenshots/Recordings

https://github.com/open-sauced/app/assets/20603494/f3638045-8c5e-4d63-8b95-8d7b40f58a00

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?
keep-track-of-it-derek-muller-gif

### CODE REVIEW

**Review by jpmcb — COMMENTED** [review]
Seems solid. Although the "clicking to another tab" experience is still abit strange:

tabbing-away

notice how when I click to a new tab and come back, it sends the modal _all the way back_ to the start screen. It's nice that the free text is preserved now so work isn't totally lost, but it is a jarring experience.

File: components/Workspaces/TrackedContributorsWizard/PasteContributorsStep.tsx [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+import { useLocalStorage } from "react-use";
 import Button from "components/atoms/Button/button";
 
 interface PasteContributorsStepProps {
   onBulkAddContributors: (contributors: string[]) => void;
 }
 
 export const PasteContributorsStep = ({ onBulkAddContributors }: PasteContributorsStepProps) => {
-  const [pastedInput, setPastedInput] = useState("");
+  const [pastedInput, setPastedInput] = useLocalStorage("bulk-add-contributors", "");
```
Comment (nickytonline) [inline components/Workspaces/TrackedContributorsWizard/PasteContributorsStep.tsx:9]: refactor (blocking): Add a default value for pastedInput so the TypeScript bang operator (!) is not required in the two spots below.

```suggestion
  const [pastedInput = "", setPastedInput] = useLocalStorage("bulk-add-contributors", "");
```

File: components/Workspaces/TrackedContributorsWizard/TrackedContributorsWizard.tsx [hunk 2]
```diff
@@ -29,6 +30,8 @@ export const TrackedContributorsWizard = ({ onAddToTrackingList, onCancel }: Tra
   const [searchTerm, setSearchTerm] = useState<string | undefined>();
   const { data, isError, isLoading } = useSearchContributors(searchTerm);
 
+  const [pastedInput, setPastedInput, removePastedInput] = useLocalStorage("bulk-add-contributors", "");
```
Comment (nickytonline) [inline components/Workspaces/TrackedContributorsWizard/TrackedContributorsWizard.tsx:33]: refactor (blocking): The `pastedInput` and `setPastedInput` variables from the destructured array are not used. Omit them.

```suggestion
const [, , removePastedInput] = useLocalStorage("bulk-add-repos", "");
```

**Review by nickytonline — CHANGES_REQUESTED** [review]
Tested it locally and behaves as expected. Just a couple change requests @zeucapua.

**Comment by zeucapua** [comment]
> Seems solid. Although the "clicking to another tab" experience is still abit strange:
> 
> tabbing-away [ tabbing-away ](https://private-user-images.githubusercontent.com/23109390/314275504-319be2f0-21fa-4b7e-80b1-e5f6c68a5953.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA5NTQ5NjMsIm5iZiI6MTcxMDk1NDY2MywicGF0aCI6Ii8yMzEwOTM5MC8zMTQyNzU1MDQtMzE5YmUyZjAtMjFmYS00YjdlLTgwYjEtZTVmNmM2OGE1OTUzLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzIwVDE3MTEwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYxZjRkNmEyZjZlZTRmMzQ5N2I3YThjNDllOTljMDczYjdhZTQ0YmM2YzI4M2IxNTBiMDE4ZTkzODI2ODdmMjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.byBmI0JyEJQlv0Wt8mzD58Tr1_ouirSwTiNjX_h0b34) [ ](https://private-user-images.githubusercontent.com/23109390/314275504-319be2f0-21fa-4b7e-80b1-e5f6c68a5953.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA5NTQ5NjMsIm5iZiI6MTcxMDk1NDY2MywicGF0aCI6Ii8yMzEwOTM5MC8zMTQyNzU1MDQtMzE5YmUyZjAtMjFmYS00YjdlLTgwYjEtZTVmNmM2OGE1OTUzLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzIwVDE3MTEwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYxZjRkNmEyZjZlZTRmMzQ5N2I3YThjNDllOTljMDczYjdhZTQ0YmM2YzI4M2IxNTBiMDE4ZTkzODI2ODdmMjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.byBmI0JyEJQlv0Wt8mzD58Tr1_ouirSwTiNjX_h0b34)
> 
> notice how when I click to a new tab and come back, it sends the modal _all the way back_ to the start screen. It's nice that the free text is preserved now so work isn't totally lost, but it is a jarring experience.

will be addressed with #2799

**Review by jpmcb — APPROVED** [review]

**Review by nickytonline — APPROVED** [review]

**Pull request merged**